Source/BaseTableDataDisplayManager/Support/CalculatableHeightNonReusableCellGenerator.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  CalculatableHeightNonReusableCellGenerator.swift
+//  ReactiveDataDisplayManager
+//
+//  Created by Alexander Filimonov on 02/03/2020.
+//  Copyright © 2020 Александр Кравченков. All rights reserved.
+//
+
+import Foundation
+
+/// Class for generating non-reusable Configurable UITableViewCell with calculated height
+public class CalculatableHeightNonReusableCellGenerator<Cell: CalculatableHeight>: TableCellGenerator, SelectableItem where Cell: UITableViewCell {
+
+    // MARK: - Public properties
+
+    public var didSelectEvent = BaseEvent<Void>()
+    private(set) public var model: Cell.Model
+    private(set) public lazy var cell: Cell? = {
+        return Cell.fromXib()
+    }()
+
+    // MARK: - Private Properties
+
+    private let registerType: CellRegisterType
+
+    // MARK: - Initialization
+
+    public init(with model: Cell.Model,
+                registerType: CellRegisterType = .nib) {
+        self.model = model
+        self.registerType = registerType
+    }
+
+    // MARK: - Public Methods
+
+    public func update(model: Cell.Model) {
+        self.model = model
+        cell?.configure(with: model)
+    }
+
+    // MARK: - TableCellGenerator
+
+    public var identifier: UITableViewCell.Type {
+        return Cell.self
+    }
+
+    public var cellHeight: CGFloat {
+        return Cell.getHeight(forWidth: cell?.frame.width ?? 0, with: model)
+    }
+
+    public var estimatedCellHeight: CGFloat? {
+        return Cell.getHeight(forWidth: cell?.frame.width ?? 0, with: model)
```

Comment:
Will be there some problems with case, when at the first call, before the cell is added to the table, the cell frame will be taken from the .xib file? Can this frame be irrelevant and the obtained height doesn't correspond to reality? 

Instead of the `CalculatebleHeightCellGenerator`, where cellWidth parameter provides into the init method... Think about it )

Reply:
So, the main idea is to calculate manually height according to the model. So it doesn't matter if we add cell to table or not. Because in method `getHeight` there are no frame using - only manual calculations. 

So the initialiser couldn't return some value, while static method `getHeight` can.

Reply:
I'm not talking about that)
For example. A cell in xib with a width of 320. The screen is 415 wide. The static method needs a width. The width of the lazily created cell, rather than the actual width of the screen, is input to the method. What is the probability that it will not be 415, but 320, which will lead to incorrect calculations?

Reply:
Oh, thanks, that is a good case. I will think about it